### ISSUE TITLE
Test the gradient descent C++

### ISSUE THREAD

**Issue by yotam180** [yotam180/xAI#19]
Just test or rewrite the SGD algorithm (that uses the newly finished `backprop` funcion)

**Comment by yotam180** [comment]
Works.
